File: rhine/src/main/java/com/qingmei2/rhine/di/module/AppModule.java
package com.qingmei2.rhine.di.module;


import com.qingmei2.rhine.base.BaseApplication;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Created by QingMei on 2017/8/14.
 * desc:
 */
@Module
public class AppModule {

    private BaseApplication application;

    public AppModule(BaseApplication application) {
        this.application = application;
    }

    @Singleton
    @Provides
    public BaseApplication provideApplication() {
        return application;
    }

}

File: rhine/src/main/java/com/qingmei2/rhine/di/module/AppModule.kt
package com.qingmei2.rhine.di.module


import com.qingmei2.rhine.base.BaseApplication

import javax.inject.Singleton

import dagger.Module
import dagger.Provides

/**
 * Created by QingMei on 2017/8/14.
 * desc:
 */
@Module
class AppModule(private val application: BaseApplication) {

    @Singleton
    @Provides
    fun provideApplication(): BaseApplication {
        return application
    }

}
